### ISSUE TITLE
Insert permissions interrupts table modification

### ISSUE THREAD

**Issue by dilame** [hasura/graphql-engine#3910]
As you should know, insert permissions create view based on guarded table in `hdb_views`.  That makes impossible column removing or data type change. Thats very annoying, but not the biggest issue.
I need to modify table and such restrictions should not stop me, so i run `ALTER TABLE xxx REMOVE COLUMN xxx CASCADE`, it removes column and dependend view.
So the permission is broken now. That is more annoying, but not the biggest issue.
 I need that permission. So i'm trying to modify it hoping hasura will restore it. But getting error

```
{
    "statement": "DROP VIEW \"hdb_views\".\"user__insert__bux__execution\"",
    "prepared": false,
    "error": {
        "exec_status": "FatalError",
        "hint": null,
        "message": "view \"user__insert__bux__execution\" does not exist",
        "status_code": "42P01",
        "description": null
    },
    "arguments": []
}
```
Please, just add `IF EXISTS`, it's so easy.

But that's not the biggest issue. 
Turns out i cant even remove this permission. The same error. Don't know how to move on.

**Comment by dilame** [comment]
For those who faced the same issue - go to  `hdb_catalog -> hdb_permission` find and delete related row, then in the console go to settings -> reload metadata. Now you can recreate permission.

**Comment by 0x777** [comment]
@dilame We stopped using views for enforcing insert permissions since `v1.1.0`. Can you please upgrade and let us know!

**Comment by dilame** [comment]
Wow, that's nice update! I've successfully updated, but corresponding view and routine still remained in hdb_views. Then i tried to delete the permission in console - success, but view and routine remains. I've deleted those manually in DB, so not a big problem for me, but i think for someone this could be a problem.

Also i've noticed now i can manually edit permissions in console, i mean input field below `with custom check` checkbox. Thats cool!

Hasura is something incredible. Our world don't deserve Hasura. Thank you for your work, guys!

BTW Sorry for offtop, is there any plan to make subscriptions more realtime? Maybe allow developer to send data from routines directly to clients using `NOTIFY`? Is there any plans to support remote schema subscriptions?

And last but not least - why only routines returning `SETOF` supported? It's common case to use routine for returning exactly 1 row. But on the client side i recieve an array with 1 object :( Together with views that doestn't support `_by_pk` it is impossible to get 1 cutom row on the client, need to unnest the array everytime.
